keps/168-2-pending-workloads-visibility/README.md

```diff
@@ -0,0 +1,292 @@
+
+#### Delay
+
+Additionally, in the previous proposal, Kueue updated the most prioritized workloads every 5 seconds. It is configurable, but since computing the most prioritized workloads can be expensive, it cannot be significantly reduced. Users can experience delays, which might not be convenient.
+
+
+### Goals
+
+- Support listing in order all pending workloads in a ClusterQueue, no matter the size of the queue, and without delay,
+- Support listing in order all pending workloads in a LocalQueue, no matter the size of the queue, and without delay,
+- Support fetching information about a specific workload, no matter its position in the queue, and without delay,
+- Provide consitent data across all the LocalQueues without hitting `QPS`.
+
+### Non-Goals
+
+- Provide ETA for a workload,
+- Provide information on whether a workload is admissible,
+- Provide information about the requested resource for a workload.
+
+## Proposal
+
+- Add new API,
+- Add a new Extension API Server,
+- Modify existing KueueManager's cache to support fetching additional information about pending workloads.
+
+### User Stories
+
+#### Story 1 
+
+As a user of Kueue with LocalQueue visibility only, I would like to know the position in the ClusterQueue of a workload that I've just submitted, no matter how big the queue is. Knowing the position and assuming stable velocity in the ClusterQueue, would allow me to estimate the arrival time of my workload.
+
+Provided by the [LocalQueue endpoint](#list-all-pending-workloads-in-localqueue) and the [Workload endpoint](#fetch-information-about-single-workload).
+
+#### Story 2 
+
+As an administrator of Kueue with ClusterQueue visibility, I would like to be able to check directly and compare the positions of pending workloads in the queue, no matter the size of it. It is important that data across all LocalQueue is consistent, and no two workloads have the same position in ClusterQueue. This will help me answer users' questions about their workloads.
+
+Provided by the [ClusterQueue endpoint](#list-all-pending-workloads-in-clusterqueue).
+
+#### Story 3 
+
+As a developer who uses Kueue, I would like to be able to monitor the state of my ClusterQueue/LocalQueue using dashboards. I need a mechanism that allows me to easily build it.
+
+Provided by the [ClusterQueue endpoint](#list-all-pending-workloads-in-clusterqueue) and the [LocalQueue endpoint](#list-all-pending-workloads-in-localqueue).
+
+
+### Risks and Mitigations
+
+#### DDoS
+
+The only risk we foresee is that the server may be exposed to DDoS attacks. A potential attacker may flood the server with requests, which will result in constantly locking the Kueue Manager. To mitigate this risk, we plan to introduce throttling so that even with numerous requests, the Kueue Manager remains functional. 
```

Comment:
```suggestion
One risk we foresee is that the server may be exposed to DDoS attacks. A potential attacker may flood the server with requests, which will result in constantly locking the Kueue Manager. To mitigate this risk, we plan to introduce throttling so that even with numerous requests, the Kueue Manager remains functional. 
```
To avoid the need to change this section when another risk is reported :).